pyro/infer/autoguide/guides.py

```diff
@@ -43,8 +44,8 @@ class AutoGuide(PyroModule):
     :class:`AutoGuideList` object.
 
     :param callable model: a pyro model
-    :param str prefix: a prefix that will be prefixed to all param internal sites
-        as part of the module name.
+    :param str prefix: Name of module, which is used as a prefix
```

Comment:
I was looking forward to `prefix` being removed from the interface. Do you think we could remove it in this PR, or is that unrelated? IIUC it suffices to set `self._pyro_name = "auto"` in `AutoGuide.__init__()`.

Reply:
We can remove it easily and set it to the class name or "auto". I'll push an update removing that.

Reply:
Or maybe even the name "guide"?

Reply:
Done. I was considering converting into snakecase, but it will only add further complication to the test cases / assertions elsewhere, so I have kept it `type(self).__name__`